utils/schemed-kafka-components-utils/src/main/scala/pl/touk/nussknacker/engine/schemedkafka/schemaregistry/azure/SchemaNameTopicMatchStrategy.scala

```diff
@@ -1,60 +1,96 @@
 package pl.touk.nussknacker.engine.schemedkafka.schemaregistry.azure
 
-import com.google.common.base.CaseFormat
+import pl.touk.nussknacker.engine.schemedkafka.schemaregistry.azure.SchemaNameTopicMatchStrategy.{
+  FullSchemaNameDecomposed,
+  toSchemaNamingConvention
+}
 
 // TODO: It probable should be configurable: e.g would be nice to have possibility to define static topic -> schemaName map in config
-//       Thanks to that would be possible to use existing schemas that not follow our convention in Nussknacker.
+//       Thanks to that it would be possible to use existing schemas that not follow our convention in Nussknacker.
 //       Also in case of ambiguity (>1 schemas with only different namespaces), we could pick the correct one schema.
+class SchemaNameTopicMatchStrategy(referenceTopicList: List[String]) {
+
+  /**
+    * List all reference topics matching schema names.
+    */
+  def matchAllTopics(fullSchemaNames: List[String], isKey: Boolean): List[String] = {
```

Comment:
Based on the name of the method it looks like produces schema names, but it produces topics. The best would be to introduce value class for topic name and full schema name, but we can start with more explicit method name. My proposition is `getAllMatchingTopics`

Reply:
fixed